nemo_skills/inference/model/tool_call.py

```diff
@@ -59,10 +64,28 @@ def __init__(
             overrides=tool_overrides or {},
             context=additional_config,
         )
-        # Use sensible defaults for adapters in module-based mode
-        self.schema_adapter = OpenAISchemaAdapter()
-        self.call_interpreter = ChatCompletionCallInterpreter()
-        self.response_formatter = ChatCompletionResponseFormatter()
+
+        # Detect model type and set up appropriate adapters
+        self._setup_adapters()
+
+    def _setup_adapters(self):
+        """Set up adapters based on model type."""
+        # Import here to avoid circular imports
+        from .responses import ResponsesModel
```

Comment:
what exactly is being circularly imported?

Reply:
I thought there would be a circular import due to nemo_skills.inference.model importing both tool_call and responses, but checking again it doesn't seem to happen either on relative or absolute imports, so this can be moved to the top

Reply:
Generally, if an imported package is only referenced inside a function call, then circular imports won't be a problem. It is only a problem when the top level parsing of the module and its functions/classes require the execution at the first import. In this case, the `_setup_adapters` will only be called at class init, which likely won't happen at import time.

Also in the case where there is a risk of circular imports, it is generally advisable to always use absolute imports instead of relative imports.